### PR TITLE
RTA: xFTP

### PR DESCRIPTION
**Pull request by nzspambot**
So this took longer than expected but here is a draft of xFTP

Added what I thought was the basic stuff for a New FTP site, did unit and int tests and an example

I've also moved some stuff around into helper.psm1 as it is now shared across upto 3 resources.

Is this ok? is Helper where this should be?

Take this at this point as draft and subject to change; after if you are happy where this is at the moment

thanks

---

This change is [Reviewable](https://reviewable.io/reviews/powershell/xwebadministration/202)

### CODE REVIEW

**Comment by mbreakey3** [comment]
@nzspambot  - Any updates on this?

File: Tests/Integration/MSFT_xFTP.Integration.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,169 @@
+﻿$script:DSCModuleName   = 'xWebAdministration'
```
Comment (eshaparmar) [inline Tests/Integration/MSFT_xFTP.Integration.Tests.ps1]: Can you add a description of the scenario this Int test is testing?

File: Tests/Unit/MSFT_xFTP.test.ps1 [hunk 2]
```diff
@@ -0,0 +1,2257 @@
+﻿
```
Comment (eshaparmar) [inline Tests/Unit/MSFT_xFTP.test.ps1]: Add a description as in uber tests that this test script is testing

File: Tests/Unit/MSFT_xFTP.test.ps1 [hunk 3]
```diff
@@ -0,0 +1,2257 @@
...
+                $Result = Test-TargetResource `
+                            -Ensure $MockParameters.Ensure `
+                            -Name $MockParameters.Name
+
+                It 'should return False' {
+                    $Result | Should Be $false
+                }
+            }
+
+            Context 'Check PhysicalPath is different' {
```
Comment (eshaparmar) [inline Tests/Unit/MSFT_xFTP.test.ps1]: I am not sure what the word 'different' means wrt these tests. i would change it to something more concrete. 
Same comment for the other context descriptions following this one

**Comment by johlju** [comment]
@nzspambot Impressive work on this PR! I reopen this PR since it is part of an open issue. If you still want to work on this PR I can do a review from scratch on this PR, I will label this as abandoned until I hear back from you, if not I hope someone will continue your work. Thanks for your contribution!

**Comment by t3mi** [comment]
@nzspambot 
I hope you don't mind if I continue where you left on this.

**Pull request closed without merging**